src/components/include/cbuf.h

```diff
@@ -343,121 +350,198 @@ PERCPU_EXTERN(cbuf_alloc_freelists);
 static inline struct cbuf_meta *
 __cbuf_freelist_get(int size)
 {
+	/* We have per-core, per-component, per-size free-list */
 	struct cbuf_freelist *fl = PERCPU_GET(cbuf_alloc_freelists);
 	int order = ones(size-1) - PAGE_ORDER;
 	struct cbuf_meta *m;
+	if (!(pow2(size) && size >= PAGE_SIZE)) {
+		printc("cbuf free size %d spd %d\n", size, cos_spd_id());
+	}
 	assert(pow2(size) && size >= PAGE_SIZE);
 	assert(order >= 0 && order < WORD_SIZE-PAGE_ORDER);
 	m = &(fl->freelist_head[order]);
-	if (unlikely(!m->next)) m->next = m;
+	assert(m->next);
```

Comment:
what does the last node in the freelist point to for next? I'm not sure why m->next is asserted here.